src/skeleplex/graph/modify_graph.py

```diff
@@ -0,0 +1,360 @@
+import numpy as np
+import networkx as nx
+from typing import List, Tuple, Dict, Any
+from skeleplex.graph.skeleton_graph import SkeletonGraph, get_next_node_key
+import warnings
+import logging
+from skeleplex.graph.spline import B3Spline
+from skeleplex.graph.constants import (
+    EDGE_COORDINATES_KEY,
+    EDGE_SPLINE_KEY,
+    NODE_COORDINATE_KEY,
+    GENERATION_KEY,
+    VALIDATED_KEY,
+    START_NODE_KEY,
+    END_NODE_KEY,
+    LENGTH_KEY
+
+)
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+def merge_edge(graph:nx.DiGraph, n1:int, v1:int, n2:int):
+    """merge edges in graph and add edge attributes. 
+    n1 is merged with n2. v1 is removed.
+    """
+    graph = graph.copy()
+
+    start_node = graph.nodes(data=True)[n1][NODE_COORDINATE_KEY]
+    end_node = graph.nodes(data=True)[n2][NODE_COORDINATE_KEY]
+    middle_node = graph.nodes(data=True)[v1][NODE_COORDINATE_KEY]
+
+
+    edge_attributes1 = graph.get_edge_data(n1,v1)
+    edge_attributes2 = graph.get_edge_data(v1,n2)
+    graph.remove_edge(n1,v1)
+    graph.remove_edge(v1,n2)
+    graph.remove_node(v1)
+    merge_edge = (n1,n2)
+    merge_attributes = {}
+    for key in edge_attributes1:
+        if key == 'validated':
+            if edge_attributes1[key] and edge_attributes2[key] == True:
+                merge_attributes[key] = True
+            else:
+                merge_attributes[key] = False
+
+
+        if key == EDGE_SPLINE_KEY:
+
+            points1 = edge_attributes1[EDGE_COORDINATES_KEY]
+            points2 = edge_attributes2[EDGE_COORDINATES_KEY]
+
+            #start and end node coordinates
+            #this checking is probably not necessary as we use directed graphs
+            #but just to be sure
+            if (np.allclose(points1[0], start_node) & 
+                np.allclose(points2[0], middle_node)):
+                logger.info('no flip')
+                spline_points = np.vstack((points1, points2))
+                
+            elif (np.allclose(points1[-1], start_node) &
+                   np.allclose(points2[0], middle_node)):
+                logger.info('flip 1')
+                spline_points = np.vstack((np.flip(points1, axis = 0), points2))
+            elif (np.allclose(points1[0], start_node) & 
+                  np.allclose(points2[-1], middle_node)):
+                logger.info('flip 2')
+                spline_points = np.vstack((points1, np.flip(points2, axis = 0)))
+            elif (np.allclose(points1[-1], start_node) & 
+                  np.allclose(points2[-1], middle_node)):
+                logger.info('flip both')
```

Comment:
I would consider adding the edge key to the logger info so it is more clear what is going on.

Reply:
Done